java/androidpayload/app/src/com/metasploit/stage/Payload.java

```diff
@@ -177,6 +177,7 @@ private static void readAndRunStage(DataInputStream in, OutputStream out, String
         myClass.getMethod("start",
                 new Class[]{DataInputStream.class, OutputStream.class, String[].class})
                 .invoke(stage, in, out, parameters);
-        System.exit(0);
+
+        session_expiry = -1;
```

Comment:
@OJ, I hope this doesn't break anything. This should only be reached when the user cleanly exits the session (e.g ```meterpreter > exit```)